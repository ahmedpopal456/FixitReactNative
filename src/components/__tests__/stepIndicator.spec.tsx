import React from 'react';
import 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import StepIndicator from '../stepIndicator';

describe('Step Indicator', () => {
  const props = {
    numberSteps: 3,
    currentStep: 1,
  };

  it('renders correctly', async () => {
    const stepIndicator = renderer.create(
      <StepIndicator {...props} />,
    ).toJSON();
    expect(stepIndicator).toMatchSnapshot();
  });

  it('should return proper array for step indicators', () => {
    const wrapper = shallow(
      <StepIndicator {...props} />,
    );
    expect(wrapper.instance().createStepIndicators()).toHaveLength(3);
  });
});
