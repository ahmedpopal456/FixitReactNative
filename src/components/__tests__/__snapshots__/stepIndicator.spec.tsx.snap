// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step Indicator renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "flex": 1,
      "flexDirection": "row",
      "height": 20,
      "left": 0,
      "padding": 10,
      "paddingLeft": 18,
      "paddingRight": 18,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFD14A",
        "flexGrow": 1,
        "height": 2,
        "margin": 2,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#8B8B8B",
        "flexGrow": 1,
        "height": 2,
        "margin": 2,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#8B8B8B",
        "flexGrow": 1,
        "height": 2,
        "margin": 2,
      }
    }
  />
</View>
`;
